### ISSUE TITLE
Live Polling for a rate limiter page breaks the chart

### ISSUE THREAD

**Issue by geoffharcourt** [mperham/sidekiq#4224]
Ruby version:
Sidekiq / Pro / Enterprise version(s):5.27 / 4.0.5 / 1.8.1

When we click "Live Poll" when looking at the page for a rate limiter, the chart disappears. The issue appears to be that the requests are made with a Content-Type of `text/html`, and so the return page (which looks like the whole web UI's HTML) replaces the page and kills off the chart that displayed prior to polling.

https://github.com/mperham/sidekiq/blob/master/Changes.md
https://github.com/mperham/sidekiq/blob/master/Pro-Changes.md
https://github.com/mperham/sidekiq/blob/master/Ent-Changes.md

**Comment by mperham** [comment]
Yep, that graph is specific to bucket limiters.

Looks like the JS needs to redraw the graph when the html reloads.  Right now it's only implicit on browser page load and looks like it doesn't redraw when the HTML content is replaced.

**Comment by mperham** [comment]
Unfortunately I'm stumped. I couldn't get the graph to work for me without a bugfix (the JS couldn't parse the data timestamps) so it's good that it's working better for me but I'm unsure why it is working for you. My fix:

```diff
+++ b/lib/sidekiq-ent/limiter/status.rb
@@ -131,7 +131,7 @@ module Sidekiq
               data_count.times do |idx|
                 time = now - (idx * increment)
                 bkt = bucket_name(time)
-                buckets << time
+                buckets << time.httpdate
                 conn.incrby(bkt, 0)
```

But for live polling, Sidekiq's JS graph infrastructure is not set up to work with live polling (the Dashboard implements its own AJAX polling) so I'd like to close this as WONTFIX.  Fair?

**Comment by geoffharcourt** [comment]
Yeah, if the chart can poll (or would in the future), I'm happy to see this go. Is it possible to make that button disappear on pages where it shouldn't be pressed?

**Comment by mperham** [comment]
Good idea, yep, I could hack something like `$("#poll_link").hide()` in the JS.
